src/Symfony/Component/Security/Core/User/ChainUserChecker.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\User;
+
+/**
+ * Supports multiple user checkers.
+ *
+ * This user checker is a collection of other user checkers
+ * and triggers each user checker in the sequence provided.
+ *
+ * @author Iltar van der Berg <ivanderberg@hostnet.nl>
+ */
+final class ChainUserChecker implements UserCheckerInterface
```

Comment:
I hope that by making it final, people will wonder why and ask (on IRC). We can then explain why it's final and help the developer with his design and make suggestions on what to do.